### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Zer0-bit**
- Additional startup checks( ie brew switch ON = system enters fail state).
- Refactored descale mode(needs more testing re time vs water tank levels)
- Refactored and improved temperature control based on pump flow volumes(still needs additional work but already way more promising than how it was before).
- Nextion fixes(discovered some doubling page numbers which might introduce inconsistencies)
- HOPEFULLY better boiler fill detection.
- Simplified the PIO tasks list( deleted opto).
- Lots of other fixes.

### CODE REVIEW

File: src/gaggiuino.ino [hunk 1]
```diff
@@ -134,10 +134,10 @@ static void sensorsReadPressure(void) {
     previousSmoothedPressure = currentState.smoothedPressure;
     currentState.pressure = getPressure();
     currentState.smoothedPressure = smoothPressure.updateEstimate(currentState.pressure);
-    currentState.isPressureRising = isPressureRaising();
+    currentState.isPressureRising = currentState.smoothedPressure >= previousSmoothedPressure + 0.05f;
     currentState.isPressureRisingFast = currentState.smoothedPressure >= previousSmoothedPressure + 1.55f;
-    currentState.isPressureFalling = isPressureFalling();
-    currentState.isPressureFallingFast = isPressureFallingFast();
+    currentState.isPressureFalling = currentState.smoothedPressure >= previousSmoothedPressure - 0.05f;
+    currentState.isPressureFallingFast = currentState.smoothedPressure >= previousSmoothedPressure - 0.1f;
```
Comment (judahrand) [inline src/gaggiuino.ino:140]: Inequality is the wrong way around here

File: src/gaggiuino.ino [hunk 2]
```diff
@@ -683,13 +683,12 @@ bool isBoilerFillPhase(unsigned long elapsedTime) {
 
 bool isBoilerFull(unsigned long elapsedTime) {
   bool boilerFull = false;
-  if (elapsedTime > BOILER_FILL_START_TIME + 1000) {
+  if (elapsedTime > BOILER_FILL_START_TIME + 1000UL) {
     boilerFull =  (previousSmoothedPressure - currentState.smoothedPressure > -0.05f)
                 &&
                   (previousSmoothedPressure - currentState.smoothedPressure < 0.05f);
```
Comment (judahrand) [inline src/gaggiuino.ino:689]: Do you actively want to keep this disconnected from the definitions of `isPressureRising` and `isPressureFalling` or could this be rephrased as `boilerFull = !(currentState.isPressureRising || currentState.isPressureFalling);`?

Comment (Zer0-bit) [reply src/gaggiuino.ino:689]: Yes, these can change independently of the other definitions, mixing them up can surely create confusion later as tuning those for one feature(ex predictive scales) will fuck up the other one.

File: src/gaggiuino.ino (hunk 1 above)
Comment (Zer0-bit) [reply src/gaggiuino.ino]: Thx corrected.

File: src/functional/descale.h [hunk 3]
```diff
@@ -8,17 +8,25 @@
...
 //###############################____DESCALE__CONTROL____######################################
 //#############################################################################################
 
+typedef enum {
+  IDLE,
+  DESCALING_PHASE1,
+  DESCALING_PHASE2,
+  DESCALING_PHASE3,
+  FINISHED
+} DescalingState;
```
[the remark this answers is not in the document]
Comment (Zer0-bit) [reply src/functional/descale.h:25]: Thx for the suggestion, noted.

**Pull request merged**